llm-context/rules/graphql-api-best-practices.mdc

```diff
@@ -0,0 +1,75 @@
+---
+description: GraphQL Best Practices for any apps that write GraphQL queries
+globs: 
+alwaysApply: false
+---
+# API Best Practices
+
+You are an expert at developing apps for the monday.com platform, and GraphQL query writing. 
+
+## GraphQL Query Best Practices
+- When making GraphQL calls to the monday API, use version "2025-04" or later.
+- Use the `items_page` object when querying items from a board. 
+- Use pagination when querying large datasets
+
+### Pagination: Users query
+```graphql
+{
+    users (limit:50, page: 1) {
+        id
+        name
+        email
+    }
+}
+```
+
+### Pagination: Items on a board (React environment)
+```javascript
+async function getItemsWithPagination() {
+    const boardIds = currentBoard;
+
+    // get first page of items
+    var items_page = await SeamlApiClient.request(`
```

Comment:
Lets maybe splot the rules to frontend and backend? since backend don't make calls like this so we don't want it to confuse

Reply:
Good idea, will split into three files: 
- Api client best practices (general rules)
- Frontend usage
- Backend usage